pkg/util/provider/machinecodes/status/status.go

```diff
@@ -120,21 +119,34 @@ func FromError(err error) (s *Status, ok bool) {
 	}, false
 }
 
-// findInString need to check if this logic can be optimized
-func findInString(input string) ([]string, error) {
-	var matches []string
-
-	re := regexp.MustCompile(`\[([^\[\]]*)\]`)
-	submatchall := re.FindAllString(input, -1)
-	if submatchall == nil || len(submatchall) != 2 {
-		return nil, fmt.Errorf("Unable to decode for machine code error")
+func findCodeAndMessage(encodedMsg string) ([]string, error) {
+	var decoded []string
+	var temp []rune
+	counter := 0
+
+	for _, char := range encodedMsg {
+		switch char {
+		case '[':
+			counter++
+			temp = append(temp, char)
+		case ']':
+			if counter > 0 {
+				counter--
+				temp = append(temp, char)
+				if counter == 0 {
+					decoded = append(decoded, strings.Trim(string(temp), "[]"))
```

Comment:
I had seen a small edge case , kindly adopt this change, as it solves that.
```suggestion
					tempString := string(temp)
					decoded = append(decoded, tempString[1:len(tempString)-1])
```